packages/toolpad-app/src/toolpad/AppEditor/PageEditor/QueryEditor.tsx

```diff
@@ -235,16 +235,19 @@ function QueryNodeEditorDialog<Q>({
 
   const connectionParams = connection?.attributes.params.value;
 
-  const queryModel = React.useMemo<QueryEditorModel<any>>(
-    () => ({
+  const queryModel = React.useMemo<QueryEditorModel<any>>(() => {
+    const params =
+      (Object.entries(inputParams).filter(([, value]) => Boolean(value)) as BindableAttrEntries) ||
+      [];
+
+    return {
       query: input.attributes.query.value,
-      params:
-        (Object.entries(inputParams).filter(([, value]) =>
-          Boolean(value),
-        ) as BindableAttrEntries) || [],
-    }),
-    [input.attributes.query.value, inputParams],
-  );
+
+      // TODO mark params as @deprecated
```

Comment:
@Janpot could you help me with this - you mentioned of adding deprecation comment for old values, but I couldn't figure out where and how should we do that, maybe you have something in mind already?

Reply:
Roughly what I have in mind is something like updating the `BindingEditor` with

```tsx
<BindableEditor
  globalScope={queryScope}
  globalScopeAnotations={{
    query: {
      deprecated: true,
    },
    mySelect: {
      comment: 'Builtin select component',
      properties: {
        value: {
          comment: 'the select value...',
        },
      },
    },
  }}
/>;
```

So that we can add extra meta info to the typings and the global scope explorer.

feel free to pick that up

Reply:
> From what I understand if we want to pass custom typings then we would need to implement our own jsonToTs method? In other words we would have to manually generate typing info?

Sort of, right below we are declaring these generated types to global variables:

```tsx
    const globals = Object.keys(globalScope)
      .map((key) => `declare const ${key}: RootObject[${JSON.stringify(key)}];`)
      .join('\n');
```

We should be able to add a comment to these declarations. I guess with the current system we can only annotate the top-level keys.

Reply:
I'm trying to hardcode the annotations but I can't figure out how to make it pick up that it's deprecated 🤔 

<img width="492" alt="image" src="https://user-images.githubusercontent.com/437214/194097579-e5b3987d-e744-4e8a-a05c-b6291f365e08.png">

any ideas?

Within the app it appears without any extra details
<img width="827" alt="image" src="https://user-images.githubusercontent.com/437214/194097804-cc9ec48b-5675-4e8e-92d6-dc82c7421d44.png">

Reply:
I'm not sure whether jsdoc picks up comments like `//*`, did you try `/** @deprecated ... */`?

Reply:
Oh damn, you are right 🤦 I did the other one correctly and for whatever reason used `//*`...

<img width="571" alt="image" src="https://user-images.githubusercontent.com/437214/194100914-846d69ed-1994-4ec4-8edb-bdcf992ac977.png">

So I've got some ideas now how it can work. I could try to tackle https://github.com/mui/mui-toolpad/issues/497 if you could provide more details how you think it should behave 🤔 

I think we should be able to do the example that you suggested:

```
<BindableEditor
  globalScope={queryScope}
  globalScopeMeta={{
    query: {
      deprecated: 'Use parameters instead',
    },
    mySelect: {
      comment: 'Builtin select component',
      properties: {
        value: {
          comment: 'the select value...',
        },
      },
    },
  }}
/>;
```
or at least something similar

Reply:
> I could try to tackle https://github.com/mui/mui-toolpad/issues/497 if you could provide more details how you think it should behave

I'm not sure, I think we should start by benchmarking which information retool/appsmith provide to get an idea of a baseline.

I guess Among the features we could consider:
- group the scope variables
- make them filterable with a searchbox
- label/color them (e.g. this is a component, this is a query)
- add longer form descriptions (perhaps in tooltips), links to documentation
- show type information
- add _copy path to clipboard_ so it can be pasted in the editor
- add longer form descriptions to the nested properties
- ...?

Not necessary all at once ofcourse 🙂 , and I guess @gerdadesign may want to be involved in some parts of UX/design.